core/state-machine/src/backend.rs

```diff
@@ -88,7 +88,22 @@ pub trait Backend<H: Hasher> {
 	fn pairs(&self) -> Vec<(Vec<u8>, Vec<u8>)>;
 
 	/// Get all keys with given prefix
-	fn keys(&self, prefix: &Vec<u8>) -> Vec<Vec<u8>>;
+	fn keys(&self, prefix: &[u8]) -> Vec<Vec<u8>> {
```

Comment:
Probably out of scope for this PR, but we should consider using one of the `StorageKey` types we have lying around instead of `Vec<u8>`. And consolidate them if possible/desirable.